src/index.ts

```diff
@@ -74,6 +76,35 @@ const config = Config.fromFile("./config.toml");
   } = await initBitcoin(config);
   const ethereumParams = await initEthereum(config);
 
+  const bitcoinBalanceLookup = async () => {
+    if (bitcoinWallet) {
+      return bitcoinWallet.getNominalBalance();
+    }
+    return new Big(0);
+  };
+
+  const ethereumBalanceLookup = async (): Promise<Big> => {
```

Comment:
🔧 Generating a function here and passing it all the way down through `initialiseRate` to `TestnetMarketMaker` seems to be a bit cumbersome. 
E.g. these functions are only needed if we create a `TestnetMarketMaker`, but passed-in in any case.

I would change the function `initialiseRate` to `getRateService` and pass it in only `config`, and the wallets e.g. `{ethereumParams.ethereumWallet, bitcoinWallet}`, the function itself can then define those balance lookups.

Reply:
refactored